src/rpc/handlers/BookChanges.cpp

```diff
@@ -53,10 +54,12 @@ class BookChangesHandler
     {
         reset();
 
-        for (auto const transactions =
-                 context_.get().backend->fetchAllTransactionsInLedger(
-                     ledger.seq, context_.get().yield);
-             auto const& tx : transactions)
+        std::vector<Backend::TransactionAndMetadata> const transactions =
+            Backend::synchronousAndRetryOnTimeout(
+                [seq = ledger.seq, this](auto yield) {
+                    return backend_->fetchAllTransactionsInLedger(seq, yield);
+                });
```

Comment:
So in the publishing codepath, we want to retry on timeout, but in the regular RPC handling codepath, we don't. For a regular RPC, if the function times out, we want to just return the error. We don't have to worry about returning that error, because the timeout throws an exception, and the exception is caught and an error is returned to the client.

However, in the publishing codepath, we don't actually need to fetch the transactions at all, since we already fetched them in order to publish the transactions themselves.

Reply:
yea, this sounds reasonable - i'll see how i can change it so i still reuse code where possible.
initially this retryOnTimeout creeped in because i needed a `yield` which i now did not have anymore. It seemed like this is the way we do it in the codebase from what i have seen.

Reply:
in the publish codepath, when we call this from `ReportingETL`, we do it like this. But when making database calls in response to client requests, we have a yield context already, so we can just use that.